flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -911,9 +915,11 @@ void releaseResources(boolean canceled) {
 			if (canceled) {
 				List<StateObject> statesToDiscard = new ArrayList<>();
 
-				statesToDiscard.add(metaStateHandle);
-				statesToDiscard.addAll(miscFiles.values());
-				statesToDiscard.addAll(newSstFiles.values());
+				synchronized (this) {
```

Comment:
Yes, i agree. The key point here is to make sure the stopping of the materialization thread. Synchronization does little help here. So i prefer to remove synchronization here, what do you think?